### PR TITLE
Add Dockerfile

### LINKED ISSUE #34: Add Dockerfile
Potentially use circleci to check our docker build, maybe use for some kind of integration test? (Probably needs docker swarm, seems like circleci supports that somehow...)

### PR DESCRIPTION
**Pull request by robincw-gr**
Closes #34

### CODE REVIEW

**Comment by robincw-gr** [comment]
TODO: CircleCI ?

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,15 @@
+# build stage
+FROM golang:alpine AS build-env
+RUN apk --no-cache add build-base git bzr mercurial gcc
```
Comment (dgl) [inline Dockerfile]: Using `-mod vendor` so probably doesn't need `git bzr mercurial`....

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+ADD . /src
+WORKDIR /src
+ENV CGO_ENABLED=0
+ENV GOOS=linux
+ENV GOARCH=386
```
Comment (dgl) [inline Dockerfile]: why set these any of these? I'd prefer to have 64-bit builds if we can.

Comment (robincw-gr) [reply Dockerfile]: Legacy reasons. Correct, not needed here

File: Dockerfile (hunk 1 above)
Comment (robincw-gr) [reply Dockerfile]: True

**Pull request merged**